safe_transaction_service/history/tests/test_safe_events_indexer.py

```diff
@@ -602,3 +605,35 @@ def test_safe_events_indexer(self):
         self.assertEqual(
             InternalTxDecoded.objects.count(), expected_internal_txs_decoded
         )
+
+    def test_mark_as_processed(self):
+        # SafeEventsIndexer does not use bulk saving into database,
+        # so mark_as_processed is just a optimization but not critical
+
+        # Create transaction in db so not fetching of transaction is needed
+        for safe_event in safe_events_mock:
+            tx_hash = safe_event["transactionHash"]
+            block_hash = safe_event["blockHash"]
+            if not EthereumTx.objects.filter(tx_hash=tx_hash).exists():
+                EthereumTxFactory(tx_hash=tx_hash, block__block_hash=block_hash)
+
+        # After the first processing transactions will be cached to prevent reprocessing
+        self.assertEqual(len(self.safe_events_indexer._processed_element_cache), 0)
+        self.assertEqual(
+            len(self.safe_events_indexer.process_elements(safe_events_mock)), 28
+        )
+        self.assertEqual(len(self.safe_events_indexer._processed_element_cache), 28)
+
+        # Transactions are cached and will not be reprocessed
+        self.assertEqual(
+            len(self.safe_events_indexer.process_elements(safe_events_mock)), 0
+        )
+        self.assertEqual(
+            len(self.safe_events_indexer.process_elements(safe_events_mock)), 0
+        )
+
+        # Even if we empty the cache, events will not be reprocessed again
+        self.safe_events_indexer._processed_element_cache.clear()
+        self.assertEqual(
```

Comment:
Then we need the cache here?

Reply:
It saves processing an event and a not needed call to the database